From b3870667fc1c379c43f1fbf0a565a7099bd0a0f8 Mon Sep 17 00:00:00 2001
From: Tony Lindgren <tony@atomide.com>
Date: Tue, 3 Aug 2010 12:10:01 +0300
Subject: [PATCH 0221/1092] cbus: Fix tahvo init without cbus

Basically don't even try to use it except on Nokia boards.

Signed-off-by: Tony Lindgren <tony@atomide.com>
---
 drivers/cbus/retu.c      |  8 ++++----
 drivers/cbus/tahvo-usb.c |  4 ++++
 drivers/cbus/tahvo.c     | 10 ++++++++++
 drivers/cbus/tahvo.h     |  1 +
 4 files changed, 19 insertions(+), 4 deletions(-)

diff --git a/drivers/cbus/retu.c b/drivers/cbus/retu.c
index cf2fe65..c0e7cf3 100644
--- a/drivers/cbus/retu.c
+++ b/drivers/cbus/retu.c
@@ -66,10 +66,6 @@ struct retu_irq_handler_desc {
 
 static struct retu_irq_handler_desc retu_irq_handlers[MAX_RETU_IRQ_HANDLERS];
 
-/**
- *
- *
- */
 int retu_get_status(void)
 {
 	return retu_initialized;
@@ -441,6 +437,10 @@ static int __init retu_init(void)
 {
 	int ret = 0;
 
+	if (!(machine_is_nokia770() || machine_is_nokia_n800() ||
+		machine_is_nokia_n810() || machine_is_nokia_n810_wimax()))
+			return -ENODEV;
+
 	printk(KERN_INFO "Retu/Vilma driver initialising\n");
 
 	init_completion(&device_release);
diff --git a/drivers/cbus/tahvo-usb.c b/drivers/cbus/tahvo-usb.c
index 762671a..ef9fc37 100644
--- a/drivers/cbus/tahvo-usb.c
+++ b/drivers/cbus/tahvo-usb.c
@@ -648,6 +648,10 @@ static int tahvo_usb_probe(struct device *dev)
 	struct tahvo_usb *tu;
 	int ret;
 
+	ret = tahvo_get_status();
+	if (!ret)
+		return -ENODEV;
+
 	dev_dbg(dev, "probe\n");
 
 	/* Create driver data */
diff --git a/drivers/cbus/tahvo.c b/drivers/cbus/tahvo.c
index 51d8128..8bd5ede 100644
--- a/drivers/cbus/tahvo.c
+++ b/drivers/cbus/tahvo.c
@@ -66,6 +66,12 @@ struct tahvo_irq_handler_desc {
 
 static struct tahvo_irq_handler_desc tahvo_irq_handlers[MAX_TAHVO_IRQ_HANDLERS];
 
+int tahvo_get_status(void)
+{
+	return tahvo_initialized;
+}
+EXPORT_SYMBOL(tahvo_get_status);
+
 /**
  * tahvo_read_reg - Read a value from a register in Tahvo
  * @reg: the register to read from
@@ -400,6 +406,10 @@ static int __init tahvo_init(void)
 {
 	int ret = 0;
 
+	if (!(machine_is_nokia770() || machine_is_nokia_n800() ||
+		machine_is_nokia_n810() || machine_is_nokia_n810_wimax()))
+			return -ENODEV;
+
 	printk(KERN_INFO "Tahvo/Betty driver initialising\n");
 
 	init_completion(&device_release);
diff --git a/drivers/cbus/tahvo.h b/drivers/cbus/tahvo.h
index b7a8ee1..b4c6f50 100644
--- a/drivers/cbus/tahvo.h
+++ b/drivers/cbus/tahvo.h
@@ -39,6 +39,7 @@
 
 #define MAX_TAHVO_IRQ_HANDLERS	8
 
+int tahvo_get_status(void);
 int tahvo_read_reg(int reg);
 void tahvo_write_reg(int reg, u16 val);
 void tahvo_set_clear_reg_bits(int reg, u16 set, u16 clear);
-- 
1.7.11.2

